src/map.rs

```diff
@@ -1161,21 +1168,21 @@ where
     }
 
     fn add_count(&self, n: isize, resize_hint: Option<usize>, guard: &Guard) {
-        // TODO: implement the Java CounterCell business here
-
-        use std::cmp;
-        let mut count = match n.cmp(&0) {
-            cmp::Ordering::Greater => self.count.fetch_add(n, Ordering::SeqCst) + n,
-            cmp::Ordering::Less => self.count.fetch_sub(n.abs(), Ordering::SeqCst) - n,
-            cmp::Ordering::Equal => self.count.load(Ordering::SeqCst),
-        };
+        self.counter.add(n);
 
         // if resize_hint is None, it means the caller does not want us to consider a resize.
         // if it is Some(n), the caller saw n entries in a bin
-        if resize_hint.is_none() {
-            return;
+        match resize_hint {
```

Comment:
I'm not sure I follow why this had to change? I thought we did as the same as the Java implementation here, which also has the same counter semantics as what we're now implementing.

Reply:
When looking at [this line](https://github.com/jonhoo/flurry/blob/13db2fb33d0c7ab45205ac5bb00d9b8749d959a2/jsr166/src/ConcurrentHashMap.java#L2308) I interpreted `Some(0)` and `Some(1)` as also signifying that no resize should be considered?

Reply:
Oh, huh, yeah, you're right. How about then `resize_hint.unwrap_or(0) <= 1`?

Reply:
Looking at `if check <= 1` should only be executed if there are counter cells allocated and it couldn't update base due to contention. The previous `if resize_hint.is_none()` worked better because now I'm seeing some weird behaviour where the table is not resizing when it should. For example the `map::capacity()` test fails due to this change.